python-checks/src/main/java/org/sonar/python/checks/NeedlessPassCheck.java

```diff
@@ -49,21 +51,37 @@ public void initialize(Context context) {
       if (statements.size() <= 1) {
         return;
       }
-
       statements.stream()
         .filter(st -> st.is(PASS_STMT))
         .findFirst()
         .ifPresent(st -> {
-          var issue = ctx.addIssue(st, MESSAGE);
+          var textEdit = createRemoveStatementTextEdit(statements, st);
           var quickFix = PythonQuickFix
             .newQuickFix(QUICK_FIX_MESSAGE)
-            .addTextEdit(TextEditUtils.removeStatement(st))
+            .addTextEdit(textEdit)
             .build();
-          issue.addQuickFix(quickFix);
+          ctx.addIssue(st, MESSAGE).addQuickFix(quickFix);
         });
     });
   }
 
+  private static PythonTextEdit createRemoveStatementTextEdit(List<Statement> statements, Statement toRemove) {
+    var removeIndex = statements.indexOf(toRemove);
+    var last = removeIndex == statements.size() - 1;
+    if (last) {
+      var previous = statements.get(removeIndex - 1);
+      var removeFrom = TreeUtils.getTreeSeparatorOrLastToken(previous);
+      var removeTo = TreeUtils.getTreeSeparatorOrLastToken(toRemove);
+      return TextEditUtils.removeRange(
+        removeFrom.line(),
+        removeFrom.column(),
+        removeTo.line(),
+        removeTo.column());
+    } else {
+      return TextEditUtils.removeUntil(toRemove, statements.get(removeIndex + 1));
+    }
```

Comment:
This is out of the scope of this ticket but I was wondering if this issue would not also appear in other quick fixes: where removing a statement is not removing the blank line. Maybe at some point, this logic could be part of the utils.